components/numerictextbox/events.md

```diff
@@ -12,89 +12,103 @@ position: 20
+</TelerikNumericTextBox>
 
 @code {
-    string result;
+    private string result = string.Empty;
+
+    private decimal NumericTextBoxValue { get; set; } = 12.34m;
 
-    private void MyOnChangeHandler(object theUserInput)
+    private void OnBlurHandler()
     {
-        // the handler receives an object that you may need to cast to the type of the component
-        // if you do not provide a Value, you must provide the Type parameter to the component
-        result = string.Format("The user entered: {0}", (decimal)theUserInput);
+        result = $"BLUR fired, current value is {NumericTextBoxValue}.";
     }
 }
 ````
 
-@[template](/_contentTemplates/common/general-info.md#event-callback-can-be-async)
 
->tip The `OnChange` event is a custom event and does not interfere with bindings, so you can use it together with models and forms.
+## OnChange
+
+The `OnChange` event represents a user action - confirmation of the current value. It fires when the user presses `Enter` in the input, or when the input loses focus.
+
+The NumericTextBox is a generic component, so you must either provide a `Value`, or a type to the `T` parameter of the component.
 
 >caption Handle OnChange and use two-way binding
 
 ````CSHTML
 @result
 <br />
-model value: @theTbValue
+model value: @NumericTextBoxValue
 <br />
 
-<TelerikNumericTextBox @bind-Value="@theTbValue" OnChange="@MyOnChangeHandler"></TelerikNumericTextBox>
+<TelerikNumericTextBox @bind-Value="@NumericTextBoxValue" 
+                       OnChange="@MyOnChangeHandler">
+</TelerikNumericTextBox>
 
 @code {
-    string result;
+    private string result = string.Empty;
 
-    double theTbValue { get; set; } = 1.2345;
+    private double NumericTextBoxValue { get; set; } = 1.2345;
 
-    private void MyOnChangeHandler(object theUserInput)
+    private void MyOnChangeHandler(object userInput)
     {
         // the handler receives an object that you may need to cast to the type of the component
         // when a Value is provided, the type is taken from it
-        result = string.Format("The user entered: {0}", (double)theUserInput);
+        result = string.Format("The user entered: {0}", (double)userInput);
     }
 }
 ````
 
+@[template](/_contentTemplates/common/general-info.md#event-callback-can-be-async)
+
+>tip The `OnChange` event is a custom event and does not interfere with bindings, so you can use it together with models and forms.
+
+
 ## ValueChanged
 
-The `ValueChanged` event fires on every change (keystroke) in the input. Using this event requires one-way binding for the `Value` parameter and manual update of the value in the handler. If the value is not updated, this will effective cancel the event.
+The `ValueChanged` event fires upon every valid change in the input (for example, keystroke). Using this event requires one-way binding for the `Value` parameter and manual update of the value in the handler. If the value is not updated, this will effective cancel the event.
```

Comment:
```suggestion
The `ValueChanged` event fires upon every valid change in the input (for example, keystroke). Using this event requires one-way binding for the `Value` parameter and manual updating of the value in the handler. If the value is not updated, this will effectively cancel the event.
```